src/Xamarin.Android.Build.Tasks/Tasks/Gradle.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Android.Build.Tasks;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks
+{
+	public class Gradle : AndroidToolTask
+	{
+		public override string TaskPrefix => "GRDL";
+
+		[Required]
+		public string Command { get; set; } = string.Empty;
+
+		public string Arguments { get; set; } = string.Empty;
+
+		public string ModuleName { get; set; } = string.Empty;
+
+		public string OutputPath { get; set; } = string.Empty;
+
+		public string IntermediateOutputPath { get; set; } = string.Empty;
+
+		public string AndroidSdkDirectory { get; set; } = string.Empty;
+
+		public string JavaSdkDirectory { get; set; } = string.Empty;
+
+		string InitScriptPath => Path.Combine (IntermediateOutputPath, "net.android.init.gradle.kts");
+
+		protected override string ToolName => OS.IsWindows ? "gradlew.bat" : "gradlew";
+
+		protected override string GetWorkingDirectory ()
+		{
+			return ToolPath;
+		}
+
+		protected override string GenerateFullPathToTool ()
+		{
+			return Path.Combine (ToolPath, ToolExe);
+		}
+
+		protected override string GenerateCommandLineCommands ()
+		{
+			var cmd = new CommandLineBuilder ();
+
+			var commandArg = Command;
+			if (!string.IsNullOrEmpty (ModuleName)) {
+				commandArg = $"{ModuleName}:{commandArg}";
+			}
+			cmd.AppendSwitch (commandArg);
+
+			// If an output path is specified, use init script to set the gradle projects build directory to that path
+			if (!string.IsNullOrEmpty (OutputPath)) {
+				cmd.AppendSwitchIfNotNull ("-P", $"netAndroidBuildDirOverride={OutputPath}");
+				cmd.AppendSwitchIfNotNull ("--init-script ", InitScriptPath);
+			}
+
+			if (!string.IsNullOrEmpty (Arguments))
+				cmd.AppendSwitch (Arguments);
+
+			// Do not leave gradle daemon running
+			cmd.AppendSwitch ("--no-daemon");
```

Comment:
The problem with this is that if a project contains *multiple* `@(AndroidGradleProjectReference)`s, *each build* will take longer.

A "better" / alternative idea *may* be to have the `<Gradle/>` task take *all* of the projects, and build them *in parallel*:

```xml
<Gradle Projects="@(AndroidGradleProject) … />
```

and then use have `Gradle` inherit from `AsyncTask` a'la `<MavenDownload/>`/`<Aapt/>`/etc., execute `gradlew` from all of the projects in parallel, and then at the end finish with `gradlew --stop`.

A problem with this idea is that it assumes that each `gradlew` is "the same", which might not be true.  (…and then what?  Run `gradlew --stop` for each project separately?)

This *also* assumes that `gradlew` is located alongside each `build.gradle` separately.  That might be true for some customers, but that certainly isn't true *for us*; look at `git grep GradleWPath`, and see that it's used from three separate projects, so that we have only one gradlew in the repo, not *N*.

Reply:
The "safe" approach for now in my mind would be to keep builds running in sequence in isolated processes. I think we can look at a performance pass in a future PR as [Peppers also mentioned](https://github.com/dotnet/android/pull/9270#discussion_r1745850697), where we can support building in parallel and/or potentially re-using the daemon as well.